File: packages/zynth-templates/android/app/src/main/java/modules/PerformanceModule.kt
package {{BUNDLE_ID}}.modules

import com.zynth.kit.debug.PerformanceProfiler
import com.zynth.kit.runtime.ZynthSyncModule
import org.json.JSONObject

import com.zynth.kit.runtime.ZynthModule

class PerformanceModule : ZynthModule, ZynthSyncModule {
    override val name: String = "Performance"

    override fun call(method: String, args: Array<Any?>): JSONObject {
        return JSONObject()
    }

    override fun callSync(method: String, args: Array<Any?>): Any? {
        return when (method) {
            "getLastFrameStats" -> JSONObject(PerformanceProfiler.getFrameStats()).toString()
            else -> null
        }
    }
}

File: packages/zynth-ios/ios/ZynthKit/src/runtime/PerformanceModule.swift
import Foundation

class PerformanceModule: ZynthModule, ZynthSyncModule {
    let name: String = "Performance"

    func call(method: String, args: Any?) throws -> Any? {
        return nil
    }

    func callSync(method: String, args: Any?) throws -> Any? {
        switch method {
        case "getLastFrameStats":
            return PerformanceProfiler.shared.getFrameStats()
        default:
            return nil
        }
    }
}
